regressiontests/vet_test.go

```diff
@@ -1,3 +1,5 @@
+// +build go1.10
```

Comment:
Instead of using build tags to select tests could we use `runtime.Version()` to set up different expected values?  That way it should keep working when new versions are added.

Reply:
Done. FYI, this would have kept working even once go1.11 is out, because (from https://golang.org/pkg/go/build/#hdr-Build_Constraints)
> - "go1.1", from Go version 1.1 onward
> - "go1.2", from Go version 1.2 onward
> - "go1.3", from Go version 1.3 onward
> - "go1.4", from Go version 1.4 onward
> - "go1.5", from Go version 1.5 onward
> - "go1.6", from Go version 1.6 onward
> - "go1.7", from Go version 1.7 onward
> - "go1.8", from Go version 1.8 onward
> - "go1.9", from Go version 1.9 onward
> - "go1.10", from Go version 1.10 onward

Using `runtime.Version` has the unfortunate problem of having to do string prefix comparison, because `runtime.Version()` returns the Major.Minor.Patch....